components/bl-data-grid-component/src/use-styles.js

```diff
@@ -6,17 +6,32 @@ export const useStyles = (style, width, height) => useMemo(() => {
   return { ...style, ...validStyles, flexShrink: 0 };
 }, [style, width, height]);
 
-const validateStyles = styles => {
-  return Object.entries(styles).reduce((acc, [propertyName, value]) => {
+const validateStyles = dimensions => {
+  const validDimensions = validateDimensions(dimensions);
+
+  return Object.entries(validDimensions).reduce((acc, [propertyName, value]) => {
     if (CSS.supports(propertyName, value)) {
-      return {
-        ...acc,
-        [propertyName]: value,
-      };
+      return { ...acc, [propertyName]: value };
     }
 
-    console.error(`Property ${propertyName} has wrong value!`);
+    console.error(`Property ${ propertyName } has wrong value!`);
 
     return acc;
   }, {});
 };
+
+const validateDimensions = dimensions => {
+  return Object.entries(dimensions).reduce((acc, [propertyName, value]) => {
+    if (!value) {
+      console.error(`Property ${ propertyName } has wrong value!`);
+
+      return acc;
+    }
+
+    if (value.includes("%")) {
+      return { ...acc, [propertyName]: `${ parseFloat(value) }%` };
+    }
+
+    return { ...acc, [propertyName]: `${ parseFloat(value) }px` };
+  }, {});
+};
```

Comment:
Looks like redundant logic. Could you describe cases when it's needed?

I think `CSS.supports(propertyName, value)` it's all what we need.